### DISCUSSION TITLE
Type 'boolean' is not assignable to type 'CollectionElement<object>'.ts(2322)

### DISCUSSION THREAD

**Discussion by typescript-any** [heroui-inc/heroui#2137] [Help] [answered]
```
{!loading && isLoggedIn && (
     <DropdownItem key="logout">Logout</DropdownItem>
)}
```

**Comment by Harshal3110** [comment]
> ```
> {!loading && isLoggedIn && (
>      <DropdownItem key="logout">Logout</DropdownItem>
> )}
> ```
> 
> ### how can I solve this error
> ### ERROR: Type 'false | Element' is not assignable to type 'CollectionElement'.Type 'boolean' is not assignable to type 'CollectionElement'.ts(2322)

Please let me know if you found a solution for this. I am looking for the same.

**Reply by typescript-any** [reply, marked as answer]
> > ```
> > {!loading && isLoggedIn && (
> >      <DropdownItem key="logout">Logout</DropdownItem>
> > )}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ### how can I solve this error
> > ### ERROR: Type 'false | Element' is not assignable to type 'CollectionElement'.Type 'boolean' is not assignable to type 'CollectionElement'.ts(2322)
> 
> Please let me know if you found a solution for this. I am looking for the same.


simply add hidden prop with you condition
<DropdownItem
                        hidden={!isLoggedIn }
                           key="logout"
                           textValue="Logout"
                           onClick={logoutHandler}
                        >
                           Logout
                        </DropdownItem>

**Reply by Harshal3110** [reply]
The hidden property won't work as well. I have tried it and didn't work as expected.

**Comment by wingkwong** [comment]
If you are using collection-based component, you can't do in this way since empty / null children inside is not allowed. What you can do is add a hidden `DropdownItem` in a else case to make at least one child there.

**Reply by Harshal3110** [reply]
I just found the solution. Simply did this
![image](https://github.com/nextui-org/nextui/assets/59551484/ee20e243-7db0-48b0-8166-ff3fb4b07a7d)
